src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -292,18 +314,23 @@ public static unsafe int GetClassFactoryForTypeInternal(ComActivationContextInte
         public static unsafe int RegisterClassForTypeInternal(ComActivationContextInternal* pCxtInt)
         {
 #if FEATURE_COMINTEROP_UNMANAGED_ACTIVATION
+            if (!Marshal.IsComSupported)
+            {
+                throw new NotSupportedException(SR.NotSupported_COM);
+            }
+
             ref ComActivationContextInternal cxtInt = ref *pCxtInt;
 
             if (IsLoggingEnabled())
             {
                 Log(
 $@"{nameof(RegisterClassForTypeInternal)} arguments:
-    {cxtInt.ClassId}
-    {cxtInt.InterfaceId}
-    0x{(ulong)cxtInt.AssemblyPathBuffer:x}
-    0x{(ulong)cxtInt.AssemblyNameBuffer:x}
-    0x{(ulong)cxtInt.TypeNameBuffer:x}
-    0x{cxtInt.ClassFactoryDest.ToInt64():x}");
+        {cxtInt.ClassId}
+        {cxtInt.InterfaceId}
+        0x{(ulong)cxtInt.AssemblyPathBuffer:x}
+        0x{(ulong)cxtInt.AssemblyNameBuffer:x}
+        0x{(ulong)cxtInt.TypeNameBuffer:x}
+        0x{cxtInt.ClassFactoryDest.ToInt64():x}");
```

Comment:
The formatting changes here might be problematic - these are part of the string literal - probably better to revert these.